pontoon/base/views.py

```diff
@@ -988,50 +993,75 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def get_download_content(slug, code, relative_path):
+    """
+    Get content for the file to be downloaded.
+
+    :param str slug: Project slug.
+    :param str code: Locale code.
+    :param str relative_path: Resource path.
+    """
+    # Clone or update project repository
+    project = get_object_or_404(Project, slug=slug)
+    project.repositories.first().pull()
+
+    # Get source file path
+    locale = get_object_or_404(Locale, code__iexact=code)
+    vcs_project = VCSProject(project, locales=[locale])
+
+    source_path = relative_path
+    if relative_path.endswith('.po'):
+        source_path = relative_path + 't'
+    full_path = os.path.join(vcs_project.source_directory_path(), source_path)
+
+    # Update source file from database
+    resource_file = formats.parse(full_path, full_path)
+    translations = Translation.objects.filter(
+        entity__resource__project=project,
+        entity__resource__path=relative_path,
+        entity__obsolete=False,
+        approved=True,
+        locale=locale
+    )
+    for vcs_translation in resource_file.translations:
+        db_translations = translations.filter(entity__key=vcs_translation.key)
+        vcs_translation.update_from_db(db_translations)
+    resource_file.save(locale)
```

Comment:
There's got to be a way we can serve this up without saving it back to the filesystem. We've got all the data in memory, and we're writing and then reading it again.